src/functionlenses.jl

```diff
@@ -138,12 +138,14 @@ function set(x, ::typeof(abs), y)
     s = sign(x)
     iszero(s) ? y * one(x) : y * s
 end
+set(x, ::typeof(abs2), y) = set(x, abs, √y)
 
 set(x, ::typeof(mod2pi), y) = set(x, @optic(mod(_, 2π)), y)
 set(x, f::Base.Fix2{typeof(fld)}, y) = set(x, @optic(first(fldmod(_, f.x))), y)
 set(x, f::Base.Fix2{typeof(mod)}, y) = set(x, @optic(last(fldmod(_, f.x))), y)
 set(x, f::Base.Fix2{typeof(div)}, y) = set(x, @optic(first(divrem(_, f.x))), y)
 set(x, f::Base.Fix2{typeof(rem)}, y) = set(x, @optic(last(divrem(_, f.x))), y)
+set(x, f::Base.Fix2{typeof(mod),<:AbstractUnitRange}, y) = @set mod($x - first(f.x), length(f.x)) + first(f.x) = y
```

Comment:
Can we add many more tests for this one? One with empty ranges and 100 or so with randomly generated examples and `test_getset_laws`